src/common/TestImageUdder.h

```diff
@@ -20,10 +20,34 @@ class Udder {
   const uint16_t ImageSize = ImageWidth * ImageHeight;
   static const std::vector<uint8_t> UdderImage;
 
-  static unsigned int PixelNumber;
-  unsigned int byte, bit, index;
+  // Loop through the image once and make a vector of pixel ids
+  void cachePixels(uint16_t w, uint16_t h, ESSGeometry * geom) {
+    uint32_t old = 0;
+    for (unsigned int i = 0; i < ImageWidth * ImageHeight; i++) {
+      auto pixel = getPixel(w, h, geom);
+      if (pixel < old) {
+        PixelsAreCached = true;
+        return; // we're done
+      }
+      if (pixel != 0) {
+        CachedPixels.push_back(pixel);
+      }
+      old = pixel;
+    }
+  }
+
 
   uint32_t getPixel(uint16_t w, uint16_t h, ESSGeometry * geom) {
```

Comment:
@mortenjc, what is the bit depth for the image?

Reply:
Point taken, I'll add more comments for sure. The image has a 1-bit depth.